sympy/solvers/solvers.py

```diff
@@ -1826,19 +1719,19 @@ def _expand(p):
         # reject any result that makes any denom. affirmatively 0;
         # if in doubt, keep it
         dens = _simple_dens(f, symbols)
-        result = [s for s in result if
-                  not any(checksol(d, {symbol: s}, **flags)
+        result = [r for r in result if
+                  not any(checksol(d, r, **flags)
                           for d in dens)]
     if check:
         # keep only results if the check is not False
         result = [r for r in result if
-                  checksol(f_num, {symbol: r}, **flags) is not False]
+                  checksol(f_num, r, **flags) is not False]
     return result
```

Comment:
list of dicts in arbitrary order are returned; list is empty if there is no solution